### PR TITLE
Adds trial banner and in-app upgrade

### LINKED ISSUE #1374: Requests to stripe generated on every single click in the app
### Issue Summary

When I browse someone's availability on cal.com, my browser makes many requests to `https://m.stripe.com/6`. Those can be seen by opening the web developer tools and looking at the network tab.

This seems related to this:
https://stackoverflow.com/questions/45718026/stripe-js-making-duplicate-requests-new-requests-on-state-change

### Steps to Reproduce

1. Went to any cal.com profile
2. Open your Dev tools, go to the network tab
3. Click around in the app, checking out availability for various days
4. Every time you change day, a new request to Stripe is made

### Technical details

- Browser version: Chromium, Firefox

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

Partially Fixes #1374

- Adds banner to trial users in desktop breakpoint
- Adds endpoint to redirect to checkout directly (without going to marketing page first)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- [x] Trial banner should be visible for trial users only

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code and corrected any misspellings
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: ee/components/TrialBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+const TrialBanner = () => {
+  const { t } = useLocale();
+  const query = useMeQuery();
+  const user = query.data;
+
+  if (!user || user.plan !== "TRIAL") return null;
+
+  const trialDaysLeft = dayjs(user.createdDate)
+    .add(TRIAL_LIMIT_DAYS + 1, "day")
+    .diff(dayjs(), "day");
```
Comment (zomars) [inline ee/components/TrialBanner.tsx:18]: So the last day also shows

File: ee/lib/stripe/client.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
-import { loadStripe, Stripe } from "@stripe/stripe-js";
+import { Stripe } from "@stripe/stripe-js";
+import { loadStripe } from "@stripe/stripe-js/pure";
```
Comment (zomars) [inline ee/lib/stripe/client.ts:2]: Addressing deferred loading in #1374

File: next-i18next.config.js [hunk 3]
```diff
@@ -7,4 +7,5 @@ module.exports = {
     locales: ["en", "fr", "it", "ru", "es", "de", "pt", "ro", "nl", "pt-BR", "es-419", "ko", "ja"],
   },
   localePath: path.resolve("./public/static/locales"),
+  reloadOnPrerender: process.env.NODE_ENV !== "production",
```
Comment (zomars) [inline next-i18next.config.js:10]: So we don't have to restart the server when adding translation strings in dev.

File: playwright/trial.test.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+// Using logged in state from globalSteup
+test.use({ storageState: "playwright/artifacts/trialStorageState.json" });
+
+test("Trial banner should be visible to TRIAL users", async ({ page }) => {
+  // Try to go homepage
+  await page.goto("/");
+  // It should redirect you to the event-types page
+  await page.waitForSelector("[data-testid=event-types]");
+
+  await expect(page.locator(`[data-testid=trial-banner]`)).toBeVisible();
```
Comment (zomars) [inline playwright/trial.test.ts:12]: Trial banner should be visible

**Review by PeerRich — APPROVED** [review]
looks really good, great job!

**Pull request merged**